File: dog.js
var Dog = function(args={}){
	this.name = args.name || "Doggo";
	this.age = args.age || 0;
};

Dog.prototype.greeting = function(){
	return "Hi, my name is " + this.name + ". I am " + this.age + " years old";
}


var dog = new Dog();
console.log(dog.greeting());

// Not as many built in methods
// No implicit return
// Ends with a semicolon when finishing a statement
// You have to actually call a function

File: dog.rb
class Dog

  attr_reader :name, :age

  def initialize(args={})
    @name = args.fetch(:name, "Doggo")
    @age = args.fetch(:age, 0)
  end


  def greeting
    "Hi, my name is #{name}. I am #{age} years old."
  end

end

dog = Dog.new

puts dog.greeting
